godal.go

```diff
@@ -3877,6 +3877,44 @@ func (ds *Dataset) Grid(destPath string, switches []string, opts ...GridOption)
 	return &Dataset{majorObject{C.GDALMajorObjectH(dsRet)}}, nil
 }
 
+// Dem runs the library version of gdaldem.
+// See the gdaldem doc page to determine the valid flags/opts that can be set in switches.
+//
+// Example switches (for "hillshade", switches differ per mode):
+//
+//	[]string{"-s", "111120", "-alt", "45"}
+//
+// Creation options and driver may be set in the switches slice with
+//
+//	switches:=[]string{"-co","TILED=YES","-of","GTiff"}
+func (ds *Dataset) Dem(destPath, processingMode string, colorFilename *string, switches []string, opts ...DemOption) (*Dataset, error) {
```

Comment:
The underlying `GDALDEMProcessing()` function throws an error for:
- colorFilename == nullptr && mode == COLOR_RELIEF
- colorFilename != nullptr && mode != COLOR_RELIEF
- mode == COLOR_RELIEF and failure to open non-nil colorFilename 

Passing in a pointer here rather than an empty string will trigger all 3 errors correctly. If we change that parameter to a `string`, then we'd need to pass in a `nullptr` when `colorFilename == "" && MODE != COLOR_RELIEF` in order to not trigger the 2nd error condition. We think doing this would make the behaviour of the go function more difficult to understand.

Reply:
@pericles-tpt I don't like it :)
string pointers in go have a specific semantic, and the aim of godal is to provide go-like semantics over the gdal api. It is up to you to make sure you pass in a null pointer when the user has passed in an empty string, in the likes of
```go
var cfname *C.char
if colorFilename != "" {
  cfname =  ...
  defer C.free(cfname)
}
```

Reply:
No worries, that sounds good. We just pushed the updated `Dem()` function with your recommended change.